src/zope/testrunner/tests/testrunner-subunit-v2.rst

```diff
@@ -0,0 +1,654 @@
+      !runnable
+    id=sample3.sampletests_ntd.Layer:setUp status=inprogress !runnable
+    id=sample3.sampletests_ntd.Layer:setUp status=success tags=(zope:layer)
+      !runnable
+    id=sample3.sampletests_ntd.TestSomething.test_error1 status=inprogress
+    id=sample3.sampletests_ntd.TestSomething.test_error1
+    traceback (text/x-traceback...)
+    Traceback (most recent call last):
+     testrunner-ex/sample3/sampletests_ntd.py", Line NNN, in test_error1
+        raise TypeError("Can we see errors")
+    TypeError: Can we see errors
+    <BLANKLINE>
+    id=sample3.sampletests_ntd.TestSomething.test_error1 status=fail
+      tags=(zope:layer:sample3.sampletests_ntd.Layer)
+    id=sample3.sampletests_ntd.TestSomething.test_error2 status=inprogress
+    id=sample3.sampletests_ntd.TestSomething.test_error2
+    traceback (text/x-traceback...)
+    Traceback (most recent call last):
+     testrunner-ex/sample3/sampletests_ntd.py", Line NNN, in test_error2
+        raise TypeError("I hope so")
+    TypeError: I hope so
+    <BLANKLINE>
+    id=sample3.sampletests_ntd.TestSomething.test_error2 status=fail
+      tags=(zope:layer:sample3.sampletests_ntd.Layer)
+    id=sample3.sampletests_ntd.TestSomething.test_fail1 status=inprogress
+    id=sample3.sampletests_ntd.TestSomething.test_fail1
+    traceback (text/x-traceback...)
+    Traceback (most recent call last):
+     testrunner-ex/sample3/sampletests_ntd.py", Line NNN, in test_fail1
+        self.assertEqual(1, 2)
+    AssertionError: 1 != 2
+    <BLANKLINE>
+    id=sample3.sampletests_ntd.TestSomething.test_fail1 status=fail
+      tags=(zope:layer:sample3.sampletests_ntd.Layer)
+    id=sample3.sampletests_ntd.TestSomething.test_fail2 status=inprogress
+    id=sample3.sampletests_ntd.TestSomething.test_fail2
+    traceback (text/x-traceback...)
+    Traceback (most recent call last):
+     testrunner-ex/sample3/sampletests_ntd.py", Line NNN, in test_fail2
+        self.assertEqual(1, 3)
+    AssertionError: 1 != 3
+    <BLANKLINE>
+    id=sample3.sampletests_ntd.TestSomething.test_fail2 status=fail
+      tags=(zope:layer:sample3.sampletests_ntd.Layer)
+    id=sample3.sampletests_ntd.TestSomething.test_something status=inprogress
+    id=sample3.sampletests_ntd.TestSomething.test_something status=success
+      tags=(zope:layer:sample3.sampletests_ntd.Layer)
+    id=sample3.sampletests_ntd.TestSomething.test_something_else
+      status=inprogress
+    id=sample3.sampletests_ntd.TestSomething.test_something_else status=success
+      tags=(zope:layer:sample3.sampletests_ntd.Layer)
+    id=sample3.sampletests_ntd.Layer:tearDown status=inprogress !runnable
+    id=sample3.sampletests_ntd.Layer:tearDown !runnable
+    reason (text/plain...)
+    tearDown not supported
+    id=sample3.sampletests_ntd.Layer:tearDown status=skip tags=(zope:layer)
+      !runnable
+    True
+
+Note that debugging doesn't work when running tests in a subprocess:
```

Comment:
I agree that this is unlikely to go well.  I'm slightly reluctant to actively forbid it, though, since there might be some situation I haven't thought of in which it's a useful last resort (maybe a test breaks only in `--subunit-v2` mode for some weird reason and you need to debug it despite the likely terminal breakage).

Do you think there's a reason to actively go to the effort of forbidding this particular case, which seems like just a special case of "using `--subunit-v2` when stdout is a terminal is probably a bad idea"?

Reply:
No, I was just wondering.

Reply:
You can actually use the stdout capturing to tunnel debug sessions across machines - in principle all through the testr stack as well, though there is some missing glue to make that work completely (I need a terminal multiplexer :().

Put another way: you can drop a subunit filter onto the terminal if you're running tests in subunit mode and want to debug without seeing the protocol itself.